src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -112,15 +120,31 @@ const getQueryClient = async (
   cryptography?: CryptographyServiceSetup,
   rootClient?: Client,
   dataSourceId?: string,
+  request?: OpenSearchDashboardsRequest,
+  authRegistry?: IAuthenticationMethodRegistery,
   requireDecryption: boolean = true
 ): Promise<Client> => {
-  const {
+  let credential;
+  let {
     auth: { type },
-    endpoint,
+    name,
   } = dataSourceAttr;
+  const { endpoint } = dataSourceAttr;
+  name = name ?? type;
```

Comment:
Is this cause the duplicate in the mapping, if name is null or undefined and use type as name?

Reply:
fields are different and they server different purpose. name is identifier for authentication method. Type is the method for signing the request. Type can be no authentication, basic authentication and AWS SigV4. To support existing authentication method, I'm using type as name. But consider I want to add new auth method for role based authentication. As auth method owner, I can do assume role and retrieve access key, secret key and then pass to data source plugin to sign the request using AWS Sig4. In that case, name would 'ABC` and type would 'awssigv4'.